tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -121,13 +121,13 @@ static PyObject * THPVariable_arange(PyObject* self, PyObject* args, PyObject* k
       const auto options = TensorOptions()
           .dtype(scalarType)
           .device(r.device(4))
-          .layout(r.layout(3).layout)
+          .layout(r.layoutOptional(3))
```

Comment:
The `Layout` argument here is not optional (there is no question mark after it), so this should be a `r.layout(3)` call. The original PR gives bad guidance here, because it was also trying to change the types of the Layout to be Layout?, but that's no longer in scope for this PR.

Basically, this line should look similar to the dtype/device setters you see above. A bunch of other places need to respect in similar ways.